### PR TITLE
Task b cc

### PR DESCRIPTION
**Pull request by tarimislam**
cpp files are modified according to code convention

### CODE REVIEW

**Review by hamza-sadruddin — CHANGES_REQUESTED** [review]
In each file according to rule 13. have header guards professor has given 2 guards at end and beginning. The same follows for file subnetCadmium.hpp
example:
```
#ifndef MY_HEADER
#define MY_HEADER
... #endif // __ MY_HEADER__
```

https://github.com/navid-hossain/AlternateBitProtocolSimulator/blob/2414dec50046af0773523620b7cacc58d2321be9/include/atomics/receiverCadmium.hpp#L7-L8

File: include/atomics/subnetCadmium.hpp [hunk 1]
```diff
@@ -28,30 +28,30 @@ using namespace cadmium;
 using namespace std;
 
 //Port definition
-    struct Subnet_defs{
+    struct subnet_defs{
```
Comment (hamza-sadruddin) [inline include/atomics/subnetCadmium.hpp]: Please have it indented, to the first column.

File: include/atomics/subnetCadmium.hpp [hunk 2]
```diff
@@ -64,50 +64,56 @@ using namespace std;
...
 
             // output function
             typename make_message_bags<output_ports>::type output() const {
-              typename make_message_bags<output_ports>::type bags;
-              Message_t out;
-              if ((double)rand() / (double) RAND_MAX  < 0.95){
+                typename make_message_bags<output_ports>::type bags;
+                Message_t out;
+                if ((double)rand() / (double) RAND_MAX  < 0.95){
                 out.value = state.packet;
```
Comment (hamza-sadruddin) [inline include/atomics/subnetCadmium.hpp]: Please indent the inside block of` if `statement to 4 spaces.

**Review by hamza-sadruddin — CHANGES_REQUESTED** [review]

**Review by hamza-sadruddin — CHANGES_REQUESTED** [review]
Please use lower case letters for naming files according to the convention. eg. receiverCadmium.hpp, becomes receiver_cadmium.hpp.

File: test/src/receiver/main.cpp [hunk 3]
```diff
@@ -13,117 +13,165 @@
...
+/***APPLICATION GENERATOR****/
+/****************************/
 
-/********************************************/
-/****** APPLICATION GENERATOR *******************/
-/********************************************/
 template<typename T>
-class ApplicationGen : public iestream_input<Message_t,T> {
+class ApplicationGen : public iestream_input<message_t,T> {
 public:
```
Comment (khanyousefzai) [inline test/src/receiver/main.cpp]: check indentation

File: test/src/receiver/main.cpp [hunk 4]
```diff
@@ -13,117 +13,165 @@
...
-  ApplicationGen() = default;
-  ApplicationGen(const char* file_path) : iestream_input<Message_t,T>(file_path) {}
+    ApplicationGen() = default;
```
Comment (khanyousefzai) [inline test/src/receiver/main.cpp]: Check indentation

File: test/src/receiver/main.cpp [hunk 5]
```diff
@@ -13,117 +13,165 @@
...
+    ApplicationGen(const char* file_path) : 
+	iestream_input<message_t,T>(file_path){}
 };
 
 
 int main(){
 
-  auto start = hclock::now(); //to measure simulation execution time
+    auto start = hclock::now(); //to measure simulation execution time
```
Comment (khanyousefzai) [inline test/src/receiver/main.cpp]: Check indentation

File: test/src/receiver/main.cpp [hunk 6]
```diff
@@ -13,117 +13,165 @@
...
+/***************************/
+/*********Loggers **********/
+/***************************/
 
-/*************** Loggers *******************/
-  static std::ofstream out_data("../test/data/receiver/receiver_test_output.txt");
+    static std::ofstream out_data("../test/data/receiver/receiver_test_output.txt");
     struct oss_sink_provider{
         static std::ostream& sink(){          
-            return out_data;
```
Comment (khanyousefzai) [inline test/src/receiver/main.cpp]: Check indentation

File: test/src/receiver/main.cpp [hunk 7]
```diff
@@ -13,117 +13,165 @@
...
 cadmium::dynamic::modeling::Ports iports_TOP = {};
+
 cadmium::dynamic::modeling::Ports oports_TOP = {typeid(outp)};
+
 cadmium::dynamic::modeling::Models submodels_TOP = {generator, receiver1};
+
 cadmium::dynamic::modeling::EICs eics_TOP = {};
+
 cadmium::dynamic::modeling::EOCs eocs_TOP = {
-  cadmium::dynamic::translate::make_EOC<Receiver_defs::out,outp>("receiver1")
```
Comment (khanyousefzai) [inline test/src/receiver/main.cpp]: Check indentation

File: test/src/receiver/main.cpp [hunk 8]
```diff
@@ -13,117 +13,165 @@
...
+    cadmium::dynamic::translate::make_EOC<receiver_defs::out,outp>("receiver1")
 };
+
 cadmium::dynamic::modeling::ICs ics_TOP = {
-  cadmium::dynamic::translate::make_IC<iestream_input_defs<Message_t>::out,Receiver_defs::in>("generator","receiver1")
```
Comment (khanyousefzai) [inline test/src/receiver/main.cpp]: Check indentation

File: test/src/sender/main.cpp [hunk 9]
```diff
@@ -77,65 +94,100 @@ using logger_top=cadmium::logger::multilogger<log_messages, global_time>;
...
-    auto elapsed1 = std::chrono::duration_cast<std::chrono::duration<double, std::ratio<1>>>(hclock::now() - start).count();
-    cout << "Model Created. Elapsed time: " << elapsed1 << "sec" << endl;
-    
-    cadmium::dynamic::engine::runner<NDTime, logger_top> r(TOP, {0});
-    elapsed1 = std::chrono::duration_cast<std::chrono::duration<double, std::ratio<1>>>(hclock::now() - start).count();
-    cout << "Runner Created. Elapsed time: " << elapsed1 << "sec" << endl;
+std::shared_ptr<cadmium::dynamic::
+modeling::coupled<TIME>> TOP = std::
+make_shared<cadmium::dynamic::modeling::coupled<TIME>>(
+       "TOP",
```
Comment (khanyousefzai) [inline test/src/sender/main.cpp]: Check indentation

**Review by khanyousefzai — CHANGES_REQUESTED** [review]

File: test/src/receiver/main.cpp (hunk 3 above)
Comment (tarimislam) [reply test/src/receiver/main.cpp]: Changes have been done

File: test/src/receiver/main.cpp (hunk 4 above)
Comment (tarimislam) [reply test/src/receiver/main.cpp]: Changes have been done

File: test/src/receiver/main.cpp (hunk 8 above)
Comment (tarimislam) [reply test/src/receiver/main.cpp]: Changes have been done

File: test/src/sender/main.cpp (hunk 9 above)
Comment (tarimislam) [reply test/src/sender/main.cpp]: Changes have been done

File: test/src/receiver/main.cpp (hunk 7 above)
Comment (tarimislam) [reply test/src/receiver/main.cpp]: Changes have been done

**Review by hamza-sadruddin — APPROVED** [review]
Looks good to me.

**Review by khanyousefzai — APPROVED** [review]

**Comment by khanyousefzai** [comment]
It looks Ok.

**Pull request merged**